cmd/jaeger/integration/grpc_test.go

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package integration
+
+import (
+	"log"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/testbed/testbed"
+	"github.com/stretchr/testify/require"
+
+	"github.com/jaegertracing/jaeger/cmd/jaeger/integration/datareceivers"
```

Comment:
Do you foresee other data receivers? If not we could define this directly in integration pkg since they are closely coupled. And we could also move integration pkg under internal so that you can call components() function

Reply:
Thoughts?

Reply:
> Do you foresee other data receivers?

We will need another data receivers for Kafka, I've created one at #4971. Actually, I've just read your comment at Kafka data receiver that you prefer this package to be moved under "$root/internal/cit/". Is that still relevant?

> we could also move integration pkg under internal so that you can call components() function

I've just tried that to move `cmd/jaeger/integration` to `cmd/jaeger/internal/integration` but I can't call the components() function when it is in private. Am I miss understood something here?

Reply:
Yeah we'd still need Components as public, I suppose. Although another alternative is to define your own components for integration tests, as you probably won't need everything from production build.

Reply:
Okay, I just felt like making the Components function public, since it will still be kept private within the internal directory. Defining a different components function means maintaining two similar functions with different environments, which could lead to oversight in the future (probably? this is only my thought process).